staging/src/k8s.io/legacy-cloud-providers/aws/aws.go

```diff
@@ -1588,6 +1588,10 @@ func (c *Cloud) InstanceExistsByProviderID(ctx context.Context, providerID strin
 
 	instances, err := c.ec2.DescribeInstances(request)
 	if err != nil {
+		// if err is InstanceNotFound, return false with no error
+		if isAWSErrorInstanceNotFound(err) {
```

Comment:
Is this a behavioral change from the AWS API? It seems like before you could never get a `InstanceNotFound` error from `DescribeInstances` (plural)

Reply:
This is not a behavioral change. After a node got deleted, `DescribeInstances` shows it in terminated state for several hours. And after that `DescribeInstances` returns error with `InvalidInstanceID.NotFound `.

Reply:
Is that only if all instance IDs are terminated for several hours?

Reply:
Yes. and for this specific case, where only one instance ID is passed in, `InvalidInstanceID.NotFound` will be returned when that instance is terminated after several hours

Reply:
One more clarifying question -- is the same error returned when we fetch for only 1 instance (i.e. `DescribeInstance`) or is there a separate error value for that? 

Concerned about future changes using `isAWSErrorInstanceNotFound` assuming it also handles error from `DescribeInstance`.

Reply:
nvm I see there's no `DescribeInstance` method anyways.. thought there was for some reason